render_to_string context argument should accept Context

# Bug report

## What's wrong

```
render_test.py:8: error: Argument 2 to "render_to_string" has incompatible type "RequestContext"; expected "Optional[Dict[str, Any]]"
```

```python
link = render_to_string(link_template, RequestContext(request))
```

This is because:

https://github.com/typeddjango/django-stubs/blob/963d50c717282c340fa7d397bdb0402a4e7f9434/django-stubs/template/loader.pyi#L9-L14

## How is that should be

The `render_to_string` function simply passes the `context` argument to the `template.render` function. This function already accepts the correct Union of arguments:

https://github.com/typeddjango/django-stubs/blob/b397ec04a95e009c7f86b6bcb7325d2059384dc1/django-stubs/template/base.pyi#L62-L64

Also, `django.shortcuts.render` and `django.shortcuts.render_to_response` both should also accept the same type for context arg, and currently they do not:

https://github.com/typeddjango/django-stubs/blob/963d50c717282c340fa7d397bdb0402a4e7f9434/django-stubs/shortcuts.pyi#L9-L23

## System information

- OS: MacOS Catalina 10.15.4 (19E287)
- `python` version: 3.8.0
- `django` version: 2.2
- `mypy` version: 0.6.1
- `django-stubs` version: 1.5.0

I am taking a stab at fixing this in my fork, will submit a PR when ready

--- a/django-stubs/shortcuts.pyi
+++ b/django-stubs/shortcuts.pyi
@@ -9,10 +9,11 @@ from django.http.response import (
 
 from django.db.models import Manager, QuerySet
 from django.http import HttpRequest
+from django.template.context import Context
 
 def render_to_response(
     template_name: Union[str, Sequence[str]],
-    context: Optional[Dict[str, Any]] = ...,
+    context: Optional[Union[Context, Dict[str, Any]]] = ...,
     content_type: Optional[str] = ...,
     status: Optional[int] = ...,
     using: Optional[str] = ...,
@@ -20,7 +21,7 @@ def render_to_response(
 def render(
     request: HttpRequest,
     template_name: Union[str, Sequence[str]],
-    context: Optional[Dict[str, Any]] = ...,
+    context: Optional[Union[Context, Dict[str, Any]]] = ...,
     content_type: Optional[str] = ...,
     status: Optional[int] = ...,
     using: Optional[str] = ...,

--- a/django-stubs/template/loader.pyi
+++ b/django-stubs/template/loader.pyi
@@ -1,5 +1,6 @@
 from typing import Any, Dict, List, Optional, Union
 from . import engines as engines  # noqa: F401
+from .context import Context
 
 from django.http.request import HttpRequest
 from django.template.exceptions import TemplateDoesNotExist as TemplateDoesNotExist  # noqa: F401
@@ -8,7 +9,7 @@ def get_template(template_name: str, using: Optional[str] = ...) -> Any: ...
 def select_template(template_name_list: Union[List[str], str], using: Optional[str] = ...) -> Any: ...
 def render_to_string(
     template_name: Union[List[str], str],
-    context: Optional[Dict[str, Any]] = ...,
+    context: Optional[Union[Context, Dict[str, Any]]] = ...,
     request: Optional[HttpRequest] = ...,
     using: Optional[str] = ...,
 ) -> str: ...

--- a/test-data/typecheck/template/test_loader.yml
+++ b/test-data/typecheck/template/test_loader.yml
@@ -1,0 +1,15 @@
+-   case: loader_render_to_string_accepts_context
+    main: |
+        from django.http.request import HttpRequest
+        from django.shortcuts import render, render_to_response
+        from django.template.loader import render_to_string
+        from django.template import Context, RequestContext
+
+        request = HttpRequest()
+        request_context = RequestContext(request, {'foo': 'bar'})
+
+        render_to_string('foobar', {'foo': 'bar'})
+        render_to_string('foobar', Context({'foo': 'bar'}))
+        render_to_string('foobar', request_context)
+        render(request, '403.html', request_context, status=403)
+        render_to_response('403.html', request_context)

